applications/services/desktop/desktop_settings.c

```diff
@@ -0,0 +1,76 @@
+#include "desktop_settings.h"
+#include "desktop_settings_filename.h"
+
+#include <saved_struct.h>
+#include <storage/storage.h>
+
+#define TAG "DesktopSettings"
+
+#define DESKTOP_SETTINGS_VER_10 (10)
+#define DESKTOP_SETTINGS_VER (11)
+
+#define DESKTOP_SETTINGS_PATH INT_PATH(DESKTOP_SETTINGS_FILE_NAME)
+#define DESKTOP_SETTINGS_MAGIC (0x17)
+
+typedef struct {
+    uint8_t reserved[11];
+    DesktopSettings settings;
+} DesktopSettingsV10;
+
+// Actual size of DesktopSettings v10
+static_assert(sizeof(DesktopSettingsV10) == 1044);
+
+void desktop_settings_load(DesktopSettings* settings) {
+    furi_assert(settings);
+
+    bool success = false;
+
+    do {
+        uint8_t version;
+        if(!saved_struct_get_metadata(DESKTOP_SETTINGS_PATH, NULL, &version, NULL)) break;
+
+        if(version == DESKTOP_SETTINGS_VER) {
+            success = saved_struct_load(
+                DESKTOP_SETTINGS_PATH,
+                settings,
+                sizeof(DesktopSettings),
+                DESKTOP_SETTINGS_MAGIC,
+                DESKTOP_SETTINGS_VER);
+
+        } else if(version == DESKTOP_SETTINGS_VER_10) {
+            DesktopSettingsV10 settings_v10;
```

Comment:
increasing service's stack size for that thing is rich, maybe malloc()?

Reply:
That is one big fat struct indeed, better to malloc() it.